routes/notifications.js

```diff
@@ -0,0 +1,161 @@
+const express = require("express");
+const router = express.Router();
+const Notification = require("../models/Notification");
+const joi = require("@hapi/joi");
+
+const notificationValidation = joi.object({
+  userId: joi.string().required(),
+  type: joi.string().required(),
+  isRead: joi.boolean(),
+  date: joi.date(),
+  content: joi.string().required().max(200),
+});
+
+const updateNotificationValidation = joi.object({
+  isRead: joi.boolean().required(),
+});
+
+// http://localhost/api/notifications ///////////////////////////////////////////////////////
+// GET
+// Returns all notifications
+
+router.get("/", async (req, res) => {
+  try {
+    const notifications = await Notification.find();
+    res.status(200).json(notifications);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.get("/findById/:notificationId", async (req, res) => {
+  try {
+    const notifications = await Notification.find({
+      _id: req.params.notificationId,
+    });
+    res.status(200).json(notifications);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.get("/findByUserId/:userId", async (req, res) => {
+  try {
+    const notifications = await Notification.find({
+      userId: req.params.userId,
+    });
+    res.status(200).json(notifications);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+router.post("/", async (req, res) => {
```

Comment:
Is it possible to make this function an internal function and not an exposed endpoint? I can only see a notification being created in the backend.

Reply:
whats that mean an internal function? Like just create a separate function called makePost, and call it here for the endpoint? Notifications will need to be created from the front-end upon actions such as creating an offer, accepting or rejecting an offer. Its not used currently since those features are under work by Jackie

Reply:
Yeah, we should probably confirm what the workflow will be like for the notifications @abid-salahi @Cainuse 

Should I assume that backend will handle the creation of notifications for me? (So, for example, if a user clicks "reject offer" or something along those lines, the backend will handle the changing of the offer's status. Will it, at the same time, create a notification? Or is that something I should do on the frontend?

Reply:
In my mind, it is "the backend will handle the changing of the offer's status". At least that is how I am deciding the endpoints for Offering acceptance/ Rejections right now. 

@abid-salahi  I meant as a function e.g
function notifyUser(userId, typeOfNotificaiton) {
...
}

module.export(notifyUser)

Reply:
@jacquelineyin  yes every acceptance/rejection should technically create a new notification and send it to the specific user who had made the offering. similarly create notification to the poster when a new offering has been made to it. So in backend you'd do both update the offering status and create a notification for that update accordingly.